src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Test.Common
+{
+    public class Http2LoopbackServer : IDisposable
+    {
+        private Socket _listenSocket;
+        private Stream _connectionStream;
+        private Http2Options _options;
+        private Uri _uri;
+
+        public Http2LoopbackServer(Http2Options options)
+        {
+            _options = options;
+        }
+
+        public Uri CreateServer()
```

Comment:
You should consider combining the constructor and this "CreateServer" method into the constructor.  I don't see why it needs to be separated.

In fact, it may be better to not a public constructor pattern, but instead use a factory pattern similar to the LoopbackProxyServer:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/tests/FunctionalTests/LoopbackProxyServer.cs#L287

It also support IDisposable as well as optionally needing the options argument.

Reply:
I'll give that a try, thanks.

Reply:
This is fixed in the latest commit.